api/info/connecthandler/info_connect.go

```diff
@@ -0,0 +1,327 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package connecthandler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"connectrpc.com/connect"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/ava-labs/avalanchego/api/info"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/upgrade"
+
+	v1 "github.com/ava-labs/avalanchego/proto/pb/info/v1"
+)
+
+// NewConnectInfoService returns a ConnectRPC-compatible InfoServiceHandler
+// that delegates calls to the existing Info implementation
+func NewConnectInfoService(info *info.Info) *ConnectInfoService {
+	return &ConnectInfoService{
+		Info: info,
+	}
+}
+
+type ConnectInfoService struct {
+	*info.Info
+}
+
+// GetNodeVersion returns the semantic version, database version, RPC protocol version,
+// Git commit hash, and the list of VM versions this node is running
+func (s *ConnectInfoService) GetNodeVersion(
+	_ context.Context,
+	_ *connect.Request[emptypb.Empty],
+) (*connect.Response[v1.GetNodeVersionReply], error) {
+	var jsonReply info.GetNodeVersionReply
+	if err := s.Info.GetNodeVersion(nil, nil, &jsonReply); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, err)
+	}
+
+	// Convert VM versions map to protobuf format
+	vmVersions := make(map[string]string)
+	for id, version := range jsonReply.VMVersions {
+		vmVersions[id] = version
+	}
+
+	reply := &v1.GetNodeVersionReply{
+		Version:            jsonReply.Version,
+		DatabaseVersion:    jsonReply.DatabaseVersion,
+		RpcProtocolVersion: uint32(jsonReply.RPCProtocolVersion),
+		GitCommit:          jsonReply.GitCommit,
+		VmVersions:         vmVersions,
+	}
+
+	return connect.NewResponse(reply), nil
+}
+
+// GetNodeID returns this node's unique identifier and proof-of-possession bytes
+func (s *ConnectInfoService) GetNodeID(
+	_ context.Context,
+	_ *connect.Request[emptypb.Empty],
+) (*connect.Response[v1.GetNodeIDReply], error) {
+	var jsonReply info.GetNodeIDReply
+	if err := s.Info.GetNodeID(nil, nil, &jsonReply); err != nil {
+		return nil, connect.NewError(connect.CodeInternal, err)
+	}
+
+	nodePOP := []byte{}
+	if jsonReply.NodePOP != nil {
+		// Use Marshal to serialize ProofOfPossession to bytes
+		// MarshalJSON is not ideal here. Ideally, we would use a binary serialization method (MarshalBinary, Marshal, etc.)
+		var err error
+		nodePOP, err = jsonReply.NodePOP.MarshalJSON()
```

Comment:
Why are we using json serialization here? `ProofOfPossession` is just bytes so we could just send encoded bytes over the wire. Currently this will output:
```
{
  "nodeId": "NodeID-7Xhw2mDxuDS44j42TCB6U5579esbSt3Lg",
  "nodePop": "eyJwdWJsaWNLZXkiOiIweDkwMGM5YjExOWI1YzgyZDc4MWQ0YjQ5YmU3OGMzZmM3YWU2NWYyYjQzNWI3ZWQ5ZTNhOGI5YTAzZTQ3NWVkZmY4NmQ4YTY0ODI3ZmVjOGRiMjNhNmYyMzZhZmJmMTI3ZCIsInByb29mT2ZQb3NzZXNzaW9uIjoiMHg4YmZkNmQ0ZDIwODZiMmI4MTE1ZDhmNzJmOTQwOTVmZWZlNWE2YzA3ODc2YjJhY2NmNTFhODExYWRmNTIwZjM4OWU3NGEzZDIxNTJhNmQ5MGI1MjFlMmJlNThmZmU0NjgwNDNkYzVlYTY4YjRjNDQ0MTBlYjY3ZjhkYzI0ZjEzZWQ0ZjE5NDAwMDc2NGMwZTkyMmNkMjU0YTM1ODhhNDk2MmIxY2I0ZGI3ZGU0YmI5Y2RhOWQ5ZDRkNmIwM2YzZDIifQ=="
}
```

instead of displaying the nested struct fields